internal/tsoptions/parsinghelpers_test.go

```diff
@@ -1,41 +1,44 @@
 package tsoptions
 
 import (
+	"fmt"
 	"reflect"
 	"strings"
 	"testing"
 
 	"github.com/microsoft/typescript-go/internal/core"
 )
 
-func TestParseCompilerOptionNoMissingTristates(t *testing.T) {
+func TestParseCompilerOptionNoMissingFields(t *testing.T) {
 	t.Parallel()
-
 	var missingKeys []string
 	for _, field := range reflect.VisibleFields(reflect.TypeFor[core.CompilerOptions]()) {
-		keyName := field.Name
-		// use the JSON key from the tag, if present
-		// e.g. `json:"dog[,anythingelse]"` --> dog
-		if jsonTag, ok := field.Tag.Lookup("json"); ok {
-			keyName = strings.SplitN(jsonTag, ",", 2)[0]
-		}
-
-		isTristate := field.Type == reflect.TypeFor[core.Tristate]()
-		if isTristate {
-			// Set the field on a CompilerOptions to something other than the
-			// default (i.e. not TSUnknown), then check whether
-			// ParseCompilerOptions does actually update the value for that key.
-			testValue := core.TSTrue
+		func() {
+			defer func() {
+				if r := recover(); r != nil {
+					if strings.Contains(fmt.Sprint(r), "interface conversion") {
+						// this is a success case, amazingly
```

Comment:
I can't say I like this testing method. I would much rather go and do the work to make these functions generated at this point.

Reply:
Ya, go for it! It just wasn't sitting well with me that I'd only done Tristates. Do note though that there's another field not being converted currently, as this PR shows.

Reply:
And the panicking thing is kind of awful...that could be averted with more reflection, since one could conceivably create a zero value of the required field type, rather than this generic empty interface{}

Reply:
done...that was easier than I thought. I mean search this PR for "moduleDetectionKind", because I had to add that to the switch statement to make this test pass